### ISSUE TITLE
[a11y]: Background and foreground colors are not screen-reader friendly

### ISSUE THREAD

**Issue by CBID2** [EddieHubCommunity/RepoRater#120]
### Has this bug been raised before?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Where did you find this bug?

production

### Description

I ran an accessibility test using [Accessibility Insights](https://accessibilityinsights.io/) and it came up with the following results:

  Element has insufficient color contrast of 2.39 (foreground color: #4b5563, background color: #0f1623, font size: 9.0pt (12px), font weight: normal). Expected contrast ratio of 4.5:1

### Steps to Reproduce

N/A

### Screenshots

![accessibility issue](https://github.com/EddieHubCommunity/RepoRater/assets/105683440/e4626a25-2e10-4359-886f-ce752d15a5bb)
The red icons is where the color contrasts issue occurs. 

### Do you want to work on this issue?

Yes

### If "yes" to above, please explain how you would technically implement this

I plan on using the following advice from the test:
Use foreground color: #74859b and the original background color: #0f1623 to meet a contrast ratio of 4.81:1.
However, I will test the colors in a color analyzer app to ensure that these colors are accessible.

**Comment by eddiejaoude** [comment]
Thank you for raising, what accessibility standards are these results for?

**Comment by CBID2** [comment]
> Thank you for raising, what accessibility standards are these results for?

Hi there. Sorry for the late reply. These results are to ensure that the color contrasts adhere to [WCAG 21](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html).

**Comment by eddiejaoude** [comment]
Thanks @CBID2 👍 

> Element has insufficient color contrast of 2.39 (foreground color: `#4b5563`, background color: `#0f1623`, font size: 9.0pt (12px), font weight: normal). Expected contrast ratio of `4.5:1`

**Comment by itsharshitrwt** [comment]
Hi @eddiejaoude  I am very much new to opensource and want to contribute to this issue . If  this issue is related to color contrast i have tried changing some css properties , So I am just asking if i am  doing it correctly ?
Also  if I am  not  doing it correctly  , Can you tell me how should i contribute to this issue . It will be really helpful. 
![Screenshot 2024-01-20 205306](https://github.com/EddieHubCommunity/RepoRater/assets/143377593/d2063147-48e4-4383-8d06-93b9ffa46839)

**Comment by CBID2** [comment]
> Hi @eddiejaoude I am very much new to opensource and want to contribute to this issue . If this issue is related to color contrast i have changed colors for activity feed by changing some css properties , So I am just asking if i am doing it correctly ? Also if I am not doing it correctly , Can you tell me how should i contribute to this issue . It will be really helpful.
> 
> ![Screenshot 2024-01-20 205306](https://private-user-images.githubusercontent.com/143377593/298291809-d2063147-48e4-4383-8d06-93b9ffa46839.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU3Njg5MTcsIm5iZiI6MTcwNTc2ODYxNywicGF0aCI6Ii8xNDMzNzc1OTMvMjk4MjkxODA5LWQyMDYzMTQ3LTQ4ZTQtNDM4My04ZDA2LTkzYjlmZmE0NjgzOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyMFQxNjM2NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02OGQ4OTBiNWUyNjdiMjc5NzRhY2JlMmVmMmQyMDMxMDgxMTI1NmUyMTg4OWU1ZjU3Njk3ZDE1YmIyOTVlMzQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BwzQMRKkrhjEpX3Jnq9fs5LZXs43Z1z4AtkYk6M2zQU)

Hi @itsharshitrwt. I expressed interest in completing this issue

**Comment by CBID2** [comment]
> Thanks @itsharshitrwt for showing an interest and giving a demo of your solution. From the issue description CBID2 the author of the issue is keen on solving this issue. Please read our contributing guide, this should explain our process of how to get issues assigned, any questions let us know.
> 
> 
> 
> @CBID2 thank you for replying (please note, can you not tag me in the quoted replies, people usually remove the `@` otherwise my notifications increase further)

Hi Eddie. I did not tag you in the quoted reply. That was @itsharshitrwt.

**Comment by itsharshitrwt** [comment]
Extremely sorry Guys, I was not aware of the contributing guidelines.

**Comment by eddiejaoude** [comment]
> Extremely sorry Guys, I was not aware of the contributing guidelines.

no problem, here is the guide (please note I have updated your message with `folks`, as not everyone is a `guy` 😉 and we try to use inclusive language in EddieHub) 
https://github.com/EddieHubCommunity/RepoRater/blob/prototype/CONTRIBUTING.md

**Comment by eddiejaoude** [comment]
> Hi Eddie. I did not tag you in the quoted reply. That was itsharshitrwt.

Hi @CBID2 , here is a screenshot with an arrow explaining it more - that your reply with the quote where you include the tag - if you are still unclear please let me know

<img width="981" alt="Screenshot 2024-01-20 at 20 52 22" src="https://github.com/EddieHubCommunity/RepoRater/assets/624760/8d4c3660-43f5-4287-855c-223cdc28db66">

> I expressed interest in completing this issue

Btw @CBID2  I noticed you have another issue assigned to you, please do not block other users being assigned issues

@itsharshitrwt I have assigned it to you

**Comment by CBID2** [comment]
> > Hi Eddie. I did not tag you in the quoted reply. That was itsharshitrwt.
> 
> 
> 
> Hi @CBID2 , here is a screenshot with an arrow explaining it more - that your reply with the quote where you include the tag - if you are still unclear please let me know
> 
> 
> 
> <img width="981" alt="Screenshot 2024-01-20 at 20 52 22" src="https://github.com/EddieHubCommunity/RepoRater/assets/624760/8d4c3660-43f5-4287-855c-223cdc28db66">
> 
> 
> 
> > I expressed interest in completing this issue
> 
> 
> 
> Btw @CBID2  I noticed you have another issue assigned to you, please do not block other users being assigned issues
> 
> 
> 
> @itsharshitrwt I have assigned it to you
> 
> 
> 
> 

Thanks for clarifying Eddie. As far as the other issue, I did a PR for it already. I believe you just approved it and all it needs is the approved deployment.

**Comment by CBID2** [comment]
Hi @itsharshitrwt. Any update on this?

**Comment by itsharshitrwt** [comment]
Hi @CBID2 , I genuinely appreciate your offer of assistance. It means a lot to me. By now, I have cloned the repo from github but finding the file structure a bit difficult to understand. 
Also , Do I need to change some css properties because the issue is somewhat related to visibility ? 
Thanks again for being so supportive!

**Comment by itsharshitrwt** [comment]
Hi Folks ,  I have done some color changes  for the Activity feed , please review and tell me if its correct. 
![Screenshot 2024-01-24 160229](https://github.com/EddieHubCommunity/RepoRater/assets/143377593/db03014e-748a-4df6-ab13-65b1d2b88f8c)

**Comment by CBID2** [comment]
> Hi Folks ,  I have done some color changes  for the Activity feed , please review and tell me if its correct. 
> 
> ![Screenshot 2024-01-24 160229](https://github.com/EddieHubCommunity/RepoRater/assets/143377593/db03014e-748a-4df6-ab13-65b1d2b88f8c)
> 
> 
> 
> 
> 
> 

Do you do it as a Pull Request @itsharshitrwt?

**Comment by itsharshitrwt** [comment]
No , not now .It's just how it will look like. 
Also , asking if it is a good color to use??

**Comment by CBID2** [comment]
> No , not now .It's just how it will look like. 
> 
> Also , asking if it is a good color to use?? 

It'd be best to show this through a Pull Request @itsharshitrwt. It'll make it easier for the maintainers and other contributors to give feedback on it before it gets merged.

**Comment by itsharshitrwt** [comment]
Thanks for the blog , I will surely check it.

**Comment by itsharshitrwt** [comment]
Hi folks , I tried creating a pr but it says all checks are failed , I didn't knew about that previously , so i closed the pr.
Here is the screenshot of the error .

 
![Screenshot 2024-01-27 131732](https://github.com/EddieHubCommunity/RepoRater/assets/143377593/0c4eb102-2240-4c03-9ec6-b366c1985177)

I don't know how to fix this , can someone please help me .

**Comment by CBID2** [comment]
> Hi folks , I tried creating a pr but it says all checks are failed and (Vercel) someone need to authorize it, I didn't knew about that previously , so i closed the pr.
> 
> Here is the screenshot of the error .
> 
> 
> 
>  
> 
> ![Screenshot 2024-01-27 131732](https://github.com/EddieHubCommunity/RepoRater/assets/143377593/0c4eb102-2240-4c03-9ec6-b366c1985177)
> 
> 
> 
> I don't know how to fix this , can someone please help me .

@itsharshitrwt. It means that Eddie has to approve the deployment. You can reopen your PR.

**Comment by itsharshitrwt** [comment]
Hi @CBID2  , Does it mean  i need to create a new pr ??  I can do that .

**Comment by itsharshitrwt** [comment]
Ok @CBID2  . I just re-opened it .

**Comment by itsharshitrwt** [comment]
@CBID2 , Can you please take a quick look of my PR and check if I have done everything right till now.

**Comment by itsharshitrwt** [comment]
Hi folks ,  I have created a new pull request regarding the issue .  I closed the previous one because my system was causing some errors .  I hope this pull request is ok.

**Comment by itsharshitrwt** [comment]
Thanks @CBID2 for clarifying , I saw coderabbit's feedback . I have few questions to ask related the feedback provided.

1. In the (src) folder of the project I tried changing some tailwind colors from "gray"  to "white and green" and later I found this was not  correct  I have to  use the color  " #74859b " instead , I searched for this color to apply but this doesn't matches with any of the tailwind colors  and the nearest color matches to this was "#64748b" slate-500 .  Now , I'm quiet confused how to proceed further with this. Should I change color from "text-gray-600" to "text-slate-500" or create a custom color?

![Screenshot 2024-01-29 110422](https://github.com/EddieHubCommunity/RepoRater/assets/143377593/c429a728-d0f2-4524-bdc8-e78b6f898b61)

2. There was also a feedback regarding color analyzer ,  I didn't understand it properly , like do i need to paste a screenshot in my PR  after comparing the colors in a color analyzer ?

Thank you

**Comment by CBID2** [comment]
> Thanks @CBID2 for clarifying , I saw coderabbit's feedback . I have few questions to ask related the feedback provided.
> 
> 
> 
> 1. In the (src) folder of the project I tried changing some tailwind colors in Activity.js file  from "gray"  to "white and green" and later I found this was not  correct  I have to  use the color  " #74859b " instead , I searched for this color to apply but this doesn't matches with any of the tailwind colors  and the nearest color matches to this was "#64748b" slate-500 .  Now , I'm quiet confused how to proceed further with this. Should I change color from "text-gray-600" to "text-slate-500" or create a custom color?
> 
> 
> 
> ![Screenshot 2024-01-29 110422](https://github.com/EddieHubCommunity/RepoRater/assets/143377593/c429a728-d0f2-4524-bdc8-e78b6f898b61)
> 
> 
> 
> 2. There was also a feedback regarding color analyzer ,  I didn't understand it properly , like do i need to paste a screenshot in my PR  after comparing the colors in a color analyzer ?
> 
> 
> 
> Thank you 
> 
> 
> 
> 
> 
> 

Good question @itsharshitrwt.🤔 I’m leaning towards the second option. Let's take this conversation in the contributor's channel in EddieHub’s Discord. We can get more opinions there: https://discord.gg/znsBqz8aAn

**Comment by itsharshitrwt** [comment]
I created  a custom color (#74859b) in tailwind.config.js and used that color in my Activity.js file .

![Screenshot 2024-01-29 115852](https://github.com/EddieHubCommunity/RepoRater/assets/143377593/3fac48e0-0803-4b4f-bc1e-f7b1c619694a)


Using the color in Activity.js File.

![Screenshot 2024-01-29 115813](https://github.com/EddieHubCommunity/RepoRater/assets/143377593/1a8bd6ef-81dc-4ae1-ad64-075d90b026d9)


I hope this correct method.
